### PR TITLE
Test: visualize the range

### PR DESCRIPTION
**Pull request by fisker**
- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by fisker** [comment]
Seems little bit wrong, don't know how to fix...

**Comment by thorn0** [comment]
That's sad. I like the idea.

**Comment by fisker** [comment]
I even tried location calc function from babel repo, still not right.

**Comment by fisker** [comment]
I think I fixed it.

Someone can compare it again?

**Comment by thorn0** [comment]
Looks good.

**Comment by fisker** [comment]
@thorn0  You new tests https://github.com/prettier/prettier/pull/8420/commits/3c88507b5fc702eaff407966f6bae47e462766a2

**Comment by thorn0** [comment]
Beautiful!

**Comment by fisker** [comment]
image

@thorn0 Do we want this?

File: tests/js/cursor/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -316,12 +319,15 @@ parsers: ["babel", "typescript", "flow"]
 printWidth: 80
                                                                                 | printWidth
```
Comment (thorn0) [inline tests/js/cursor/__snapshots__/jsfmt.spec.js.snap:320]: The printWidth indicator needs to be shifted 6 characters to the right for these tests.

Comment (fisker) [reply tests/js/cursor/__snapshots__/jsfmt.spec.js.snap:320]: `.match(/^\s+1 \| /)[0].length` Big file more than `6`, I'll fix it.

**Comment by thorn0** [comment]
> Do we want this?

I don't think it's needed.

**Comment by fisker** [comment]
So excited, this works.

**Comment by lydell** [comment]
It’s such a joy to occasionally visit the Pull requests tabs in this repo and see all the great stuff you all are doing! 👍

**Comment by fisker** [comment]
A memo for someone who want refactor it, the following code works too.

```js
const { SourceCode } = require("eslint");

const locationForRange = (text, rangeStart, rangeEnd) => {
  const sourceCode = new SourceCode(text, {
    tokens: [],
    comments: [],
    range: [],
    loc: {},
  });

  const start = sourceCode.getLocFromIndex(rangeStart);
  const end = sourceCode.getLocFromIndex(rangeEnd);

  start.column += 1;
  if (start.line === end.line) {
    end.column += 1;
  }
};
```

**Pull request merged**